src/indexhm12.ts

```diff
@@ -0,0 +1,50 @@
+type GridFilterTypeEnum = 'search' | 'range' | 'values';
+
+type GridFilterValue<T> = {
```

Comment:
It would be better to implement this way:
`type GridFilterValue<T, U = T> = {}`
to have the opportunity to configure both filter fields.